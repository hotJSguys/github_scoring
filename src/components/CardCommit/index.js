import React from 'react';
import styled from 'react-emotion';

const Wrapper = styled('div')``;
const General = styled('div')``;

const CardCommit = (props) => {
  return (
    <Wrapper>
      <General>
        <a href={props.link}>{props.description}</a>
        <div>
          <img src={props.url_avatar} width="60" />
          <a href="asd">{props.name}</a>
          <span>commited {props.date}</span>
        </div>
      </General>
      <a className="browse">
        <img />
      </a>
    </Wrapper>
  );
};

export default CardCommit;
