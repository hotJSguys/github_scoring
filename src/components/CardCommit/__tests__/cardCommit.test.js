import React from 'react';
import CardCommit from '../index';

describe('CardCommit ', () => {
  const description = 'update deps';
  const url_avatar = 'https://cdn.dribbble.com/users/3460/screenshots/5578703/m-avatar-ryan.png';
  const name = 'Abramov';
  const date = '3 days ago';

  const card = shallow(
    <CardCommit
      description={description}
      url_avatar={url_avatar}
      name={name}
      date={date}
      onClick={mockCallBack}
    />
  );

  const mockCallBack = jest.fn();
  it('Snapshot', () => {
    expect(card).toMatchSnapshot();
  });

  it('check prop data', () => {
    console.log('PROPS', card.props().data);
    expect(card.props().data).toBe(data);
  });

  it('Clickable button', () => {
    card.find('.browse').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
