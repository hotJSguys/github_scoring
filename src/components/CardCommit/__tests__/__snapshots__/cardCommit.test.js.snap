// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCommit  Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardCommit
    data={
      Array [
        "description",
        "link",
        "url_avatar",
        "name",
        "date",
        "commit_number",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <a />
          <div>
            <img
              width="60"
            />
            <a
              href="asd"
            />
            <span />
          </div>
        </Styled(div)>,
        <a
          className="browse"
        >
          <img />
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <a />,
            <div>
              <img
                width="60"
              />
              <a
                href="asd"
              />
              <span />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "href": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  width="60"
                />,
                <a
                  href="asd"
                />,
                <span />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": undefined,
                  "width": "60",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "href": "asd",
                },
                "ref": null,
                "rendered": null,
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img />,
          "className": "browse",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <a />
            <div>
              <img
                width="60"
              />
              <a
                href="asd"
              />
              <span />
            </div>
          </Styled(div)>,
          <a
            className="browse"
          >
            <img />
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <a />,
              <div>
                <img
                  width="60"
                />
                <a
                  href="asd"
                />
                <span />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "href": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    width="60"
                  />,
                  <a
                    href="asd"
                  />,
                  <span />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": undefined,
                    "width": "60",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "href": "asd",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img />,
            "className": "browse",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
